pleskdistup/main.py

```diff
@@ -291,6 +292,30 @@ def read_resume_data(resume_path: PathType) -> ResumeData:
         raise
 
 
+@contextmanager
+def obtain_lock(lock_file: PathType) -> typing.Generator[bool, None, None]:
+    try:
+        lock_acquired = False
+        log.info(f"Going to obtain lockfile '{lock_file}'...")
+        # O_EXCL is used to ensure that the file is created only if it does not already exist.
+        # This guarantees that we will not acquire the lock if another process has already taken it.
+        fd = os.open(lock_file, os.O_CREAT | os.O_EXCL | os.O_WRONLY)
```

Comment:
I think, it's better to store PID in the file to help with troubleshooting ghost locks.